plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/RocksDBTransaction.java

```diff
@@ -57,6 +60,21 @@ public RocksDBTransaction(
     this.metrics = metrics;
   }
 
+  @Override
+  public Optional<byte[]> get(final SegmentIdentifier segmentId, final byte[] key)
+      throws StorageException {
+    try (final OperationTimer.TimingContext ignored = metrics.getWriteLatency().startTimer()) {
+      return Optional.ofNullable(
+          innerTx.get(columnFamilyMapper.apply(segmentId), readOptions, key));
+    } catch (final RocksDBException e) {
+      if (e.getMessage().contains(NO_SPACE_LEFT_ON_DEVICE)) {
+        logger.error(e.getMessage());
+        System.exit(0);
+      }
```

Comment:
I don't think a read can cause NO_SPACE_LEFT_ON_DEVICE 🤔

Reply:
Yes, good point. Don't need that.